{
 "cells": [
  {
   "cell_type": "markdown",
   "metadata": {
    "toc": true
   },
   "source": [
    "<h1>Table of Contents<span class=\"tocSkip\"></span></h1>\n",
    "<div class=\"toc\"><ul class=\"toc-item\"><li><span><a href=\"#Sequence-alignment-for-simplot++-analyses\" data-toc-modified-id=\"Sequence-alignment-for-simplot++-analyses-1\"><span class=\"toc-item-num\">1&nbsp;&nbsp;</span>Sequence alignment for simplot++ analyses</a></span></li><li><span><a href=\"#Phylogenetic-analysis-for-the-enFeLV-derived-region\" data-toc-modified-id=\"Phylogenetic-analysis-for-the-enFeLV-derived-region-2\"><span class=\"toc-item-num\">2&nbsp;&nbsp;</span>Phylogenetic analysis for the enFeLV derived region</a></span><ul class=\"toc-item\"><li><span><a href=\"#site-1-1192\" data-toc-modified-id=\"site-1-1192-2.1\"><span class=\"toc-item-num\">2.1&nbsp;&nbsp;</span>site 1-1192</a></span></li></ul></li><li><span><a href=\"#Phylogenetic-analysis-for-the-FeLV-A-derived-region\" data-toc-modified-id=\"Phylogenetic-analysis-for-the-FeLV-A-derived-region-3\"><span class=\"toc-item-num\">3&nbsp;&nbsp;</span>Phylogenetic analysis for the FeLV-A derived region</a></span><ul class=\"toc-item\"><li><span><a href=\"#site-1619-2550\" data-toc-modified-id=\"site-1619-2550-3.1\"><span class=\"toc-item-num\">3.1&nbsp;&nbsp;</span>site 1619-2550</a></span></li></ul></li></ul></div>"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# Sequence alignment for simplot++ analyses"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 11,
   "metadata": {
    "ExecuteTime": {
     "end_time": "2023-12-20T01:42:10.024798Z",
     "start_time": "2023-12-20T01:42:05.539397Z"
    }
   },
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "outputhat23=16\n",
      "treein = 0\n",
      "compacttree = 0\n",
      "stacksize: 8192 kb\n",
      "generating a scoring matrix for nucleotide (dist=200) ... done\n",
      "All-to-all alignment.\n",
      "    0 / 12\r",
      "    1 / 12\r",
      "    2 / 12\r",
      "    3 / 12\r",
      "    4 / 12\r",
      "    5 / 12\r",
      "    6 / 12\r",
      "    7 / 12\r",
      "    8 / 12\r",
      "    9 / 12\r",
      "   10 / 12\r",
      "tbfast-pair (nuc) Version 7.407\n",
      "alg=L, model=DNA200 (2), 2.00 (6.00), -0.10 (-0.30), noshift, amax=0.0\n",
      "0 thread(s)\n",
      "\n",
      "outputhat23=16\n",
      "Loading 'hat3.seed' ... \n",
      "done.\n",
      "Writing hat3 for iterative refinement\n",
      "generating a scoring matrix for nucleotide (dist=200) ... done\n",
      "Gap Penalty = -1.53, +0.00, +0.00\n",
      "tbutree = 1, compacttree = 0\n",
      "Constructing a UPGMA tree ... \n",
      "\r",
      "    0 / 12\r",
      "   10 / 12\n",
      "done.\n",
      "\n",
      "Progressive alignment ... \n",
      "\r",
      "STEP     1 /11 \r",
      "STEP     2 /11 \r",
      "STEP     3 /11 \r",
      "STEP     4 /11 \r",
      "STEP     5 /11 \r",
      "STEP     6 /11 \r",
      "STEP     7 /11 \r",
      "STEP     8 /11 \r",
      "STEP     9 /11 \r",
      "STEP    10 /11 \r",
      "STEP    11 /11 \n",
      "done.\n",
      "tbfast (nuc) Version 7.407\n",
      "alg=A, model=DNA200 (2), 1.53 (4.59), -0.00 (-0.00), noshift, amax=0.0\n",
      "1 thread(s)\n",
      "\n",
      "minimumweight = 0.000010\n",
      "autosubalignment = 0.000000\n",
      "nthread = 0\n",
      "randomseed = 0\n",
      "blosum 62 / kimura 200\n",
      "poffset = 0\n",
      "niter = 16\n",
      "sueff_global = 0.100000\n",
      "nadd = 16\n",
      "Loading 'hat3' ... done.\n",
      "generating a scoring matrix for nucleotide (dist=200) ... done\n",
      "\n",
      "\r",
      "    0 / 12\r",
      "   10 / 12\n",
      "Segment   1/  1    1-2552\n",
      "STEP 001-001-0  identical.   \r",
      "STEP 001-001-1  identical.   \r",
      "STEP 001-002-0  identical.   \r",
      "STEP 001-002-1  identical.   \r",
      "STEP 001-003-0  identical.   \r",
      "STEP 001-003-1  identical.   \r",
      "STEP 001-004-0  identical.   \r",
      "STEP 001-004-1  identical.   \r",
      "STEP 001-005-0  identical.   \r",
      "STEP 001-005-1  identical.   \r",
      "STEP 001-006-0  accepted.\r",
      "STEP 001-006-1  identical.   \r",
      "STEP 001-007-0  identical.   \r",
      "STEP 001-007-1  identical.   \r",
      "STEP 001-008-0  identical.   \r",
      "STEP 001-008-1  identical.   \r",
      "STEP 001-009-0  identical.   \r",
      "STEP 001-009-1  identical.   \r",
      "STEP 001-010-0  identical.   \r",
      "STEP 001-010-1  identical.   \r",
      "STEP 001-011-1  identical.   \r",
      "STEP 002-011-1  identical.   \r",
      "STEP 002-010-0  identical.   \r",
      "STEP 002-010-1  identical.   \r",
      "STEP 002-009-0  identical.   \r",
      "STEP 002-009-1  identical.   \r",
      "STEP 002-008-0  identical.   \r",
      "STEP 002-008-1  identical.   \r",
      "STEP 002-007-0  identical.   \r",
      "STEP 002-007-1  identical.   \r",
      "STEP 002-006-0  identical.   \r",
      "STEP 002-006-1  identical.   \r",
      "STEP 002-005-0  identical.   \r",
      "STEP 002-005-1  identical.   \r",
      "STEP 002-004-0  identical.   \r\n",
      "Converged.\n",
      "\n",
      "done\n",
      "dvtditr (nuc) Version 7.407\n",
      "alg=A, model=DNA200 (2), 1.53 (4.59), -0.00 (-0.00), noshift, amax=0.0\n",
      "0 thread(s)\n",
      "\n",
      "\n",
      "Strategy:\n",
      " L-INS-i (Probably most accurate, very slow)\n",
      " Iterative refinement method (<16) with LOCAL pairwise alignment information\n",
      "\n",
      "If unsure which option to use, try 'mafft --auto input > output'.\n",
      "For more information, see 'mafft --help', 'mafft --man' and the mafft page.\n",
      "\n",
      "The default gap scoring scheme has been changed in version 7.110 (2013 Oct).\n",
      "It tends to insert more gaps into gap-rich regions than previous versions.\n",
      "To disable this change, add the --leavegappyregion option.\n",
      "\n",
      "nadd =  4\n",
      "npair =  48\n",
      "nseq =  16\n",
      "nlen =  9895\n",
      "use ktuples, size=6!\n",
      "nadd = 4\n",
      "ppenalty_ex = -10\n",
      "nthread = 0\n",
      "blosum 62 / kimura 200\n",
      "sueff_global = 0.100000\n",
      "norg = 12\n",
      "njobc = 13\n",
      "generating a scoring matrix for nucleotide (dist=200) ... done\n",
      "\n",
      "\n",
      "Making a distance matrix ..\n",
      "\r",
      "    1 / 12\n",
      "done.\n",
      "\n",
      "\r",
      "0 / 4                    \r",
      "f\r",
      "1 / 4                    \r",
      "f\r",
      "2 / 4                    \r",
      "WARNING: negative branch length -0.124300, step 11-0\n",
      "f\r",
      "3 / 4                    \r",
      "f\n",
      "\n",
      "Combining ..\n",
      "0 / 4\r",
      " 0 / 4\r",
      "\r",
      "   done.                      \n",
      "\n",
      "\r",
      "   done.                      \n",
      "\n",
      "addsingle (nuc) Version 7.407\n",
      "alg=A, model=DNA200 (2), 1.53 (4.59), -0.00 (-0.00), noshift, amax=0.0\n",
      "0 thread(s)\n",
      "\n",
      "\n",
      "To keep the alignment length, 20600 letters were DELETED.\n",
      "To know the positions of deleted letters, rerun the same command with the --mapout option.\n",
      "\n",
      "Strategy:\n",
      " FFT-NS-fragment (Not tested.)\n",
      " ?\n",
      "\n",
      "If unsure which option to use, try 'mafft --auto input > output'.\n",
      "For more information, see 'mafft --help', 'mafft --man' and the mafft page.\n",
      "\n",
      "The default gap scoring scheme has been changed in version 7.110 (2013 Oct).\n",
      "It tends to insert more gaps into gap-rich regions than previous versions.\n",
      "To disable this change, add the --leavegappyregion option.\n",
      "\n"
     ]
    }
   ],
   "source": [
    "%%bash\n",
    "\n",
    "mkdir output\n",
    "\n",
    "##\n",
    "cat \\\n",
    "    data/FeLV_ref.fasta | \\\n",
    "    seqkit replace -s -p \"[^atcguATCGU]\" -r \"\" \\\n",
    "    > output/FeLV_ref.QC.fasta\n",
    "\n",
    "##\n",
    "cat \\\n",
    "    data/FeLV-B.fasta | \\\n",
    "    seqkit replace -s -p \"[^atcguATCGU]\" -r \"\" \\\n",
    "    > output/FeLV-B.QC.fasta\n",
    "\n",
    "##\n",
    "mafft --auto \\\n",
    "    output/FeLV-B.QC.fasta \\\n",
    "    > output/FeLV-B.mafft.fasta\n",
    "\n",
    "##\n",
    "mafft --auto --keeplength \\\n",
    "    --addfragments output/FeLV_ref.QC.fasta \\\n",
    "    output/FeLV-B.mafft.fasta \\\n",
    "    > output/FeLV-B.mafft.smplot.fasta"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# Phylogenetic analysis for the enFeLV derived region"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## site 1-1192"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 12,
   "metadata": {
    "ExecuteTime": {
     "end_time": "2023-12-20T01:42:43.559116Z",
     "start_time": "2023-12-20T01:42:10.026863Z"
    }
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "IQ-TREE multicore version 2.1.4-beta COVID-edition for Mac OS X 64-bit built Jun 24 2021\n",
      "Developed by Bui Quang Minh, James Barbetti, Nguyen Lam Tung,\n",
      "Olga Chernomor, Heiko Schmidt, Dominik Schrempf, Michael Woodhams.\n",
      "\n",
      "Host:    iMac.local (AVX2, FMA3, 64 GB RAM)\n",
      "Command: iqtree -s output/FeLV-B.mafft.selectB1.aln.fasta -m TESTNEW -redo -bb 1000 -nt AUTO\n",
      "Seed:    337002 (Using SPRNG - Scalable Parallel Random Number Generator)\n",
      "Time:    Wed Dec 20 10:42:11 2023\n",
      "Kernel:  AVX+FMA - auto-detect threads (12 CPU cores detected)\n",
      "\n",
      "Reading alignment file output/FeLV-B.mafft.selectB1.aln.fasta ... Fasta format detected\n",
      "Alignment most likely contains DNA/RNA sequences\n",
      "Alignment has 15 sequences with 1190 columns, 88 distinct patterns\n",
      "54 parsimony-informative, 176 singleton sites, 960 constant sites\n",
      "                        Gap/Ambiguity  Composition  p-value\n",
      "   1  A_M18247.1                9.83%    passed     18.05%\n",
      "   2  FeLV-B_ON-C__clone_1     22.27%    passed     94.62%\n",
      "   3  FeLV-B_ON-C__clone_14    22.27%    passed     94.27%\n",
      "   4  FeLV-B_ON-C_clone_13     22.27%    passed     94.27%\n",
      "   5  FeLV-B_ON-C_clone_9.1     1.43%    passed     94.33%\n",
      "   6  FeLV-B_ON-C_clone_9.14    1.43%    passed     91.98%\n",
      "   7  FeLV-B_ON-H__clone16     23.03%    passed     82.53%\n",
      "   8  FeLV-B_ON-T__clone_14    23.03%    passed     84.15%\n",
      "   9  FeLV-B_ON-T__clone_2     23.03%    passed     85.20%\n",
      "  10  FeLV-B_ON-T_clone_11      0.76%    passed     83.70%\n",
      "  11  FeLV-B_ON-T_clone_3.14   25.55%    passed     78.48%\n",
      "  12  FeLV-B_ON-T_clone_3.6    25.55%    passed     75.23%\n",
      "  13  FeLV-B_ON-T_clone_9       0.76%    passed     83.70%\n",
      "  14  Y_X00188.1               13.70%    passed     97.09%\n",
      "  15  Z_AY364318.1              0.76%    passed     82.81%\n",
      "****  TOTAL                    14.38%  0 sequences failed composition chi2 test (p-value<5%; df=3)\n",
      "NOTE: FeLV-B_ON-C_clone_13 is identical to FeLV-B_ON-C__clone_14 but kept for subsequent analysis\n",
      "NOTE: FeLV-B_ON-T_clone_9 is identical to FeLV-B_ON-T_clone_11 but kept for subsequent analysis\n",
      "\n",
      "\n",
      "Create initial parsimony tree by phylogenetic likelihood library (PLL)... 0.000 seconds\n",
      "Measuring multi-threading efficiency up to 12 CPU cores\n",
      "Increase to 10 rounds for branch lengths\n",
      "8777 trees examined\n",
      "Threads: 1 / Time: 12.001 sec / Speedup: 1.000 / Efficiency: 100% / LogL: -3212\n",
      "Threads: 2 / Time: 9.712 sec / Speedup: 1.236 / Efficiency: 62% / LogL: -3212\n",
      "Threads: 3 / Time: 9.480 sec / Speedup: 1.266 / Efficiency: 42% / LogL: -3212\n",
      "BEST NUMBER OF THREADS: 2\n",
      "\n",
      "Perform fast likelihood tree search using GTR+I+G model...\n",
      "Estimate model parameters (epsilon = 5.000)\n",
      "Perform nearest neighbor interchange...\n",
      "Estimate model parameters (epsilon = 1.000)\n",
      "1. Initial log-likelihood: -2719.853\n",
      "Optimal log-likelihood: -2719.794\n",
      "Rate parameters:  A-C: 2.76426  A-G: 17.27544  A-T: 2.77280  C-G: 1.60187  C-T: 9.40420  G-T: 1.00000\n",
      "Base frequencies:  A: 0.279  C: 0.267  G: 0.241  T: 0.213\n",
      "Proportion of invariable sites: 0.407\n",
      "Gamma shape alpha: 998.112\n",
      "Parameters optimization took 1 rounds (0.002 sec)\n",
      "Time for fast ML tree search: 0.018 seconds\n",
      "\n",
      "NOTE: ModelFinder requires 0 MB RAM!\n",
      "ModelFinder will test up to 286 DNA models (sample size: 1190) ...\n",
      " No. Model         -LnL         df  AIC          AICc         BIC\n",
      "WARNING: Number of threads seems too high for short alignments. Use -T AUTO to determine best number of threads.\n",
      "  1  GTR+F         2722.304     35  5514.608     5516.792     5692.468\n",
      "WARNING: Number of threads seems too high for short alignments. Use -T AUTO to determine best number of threads.\n",
      "  2  GTR+F+I       2719.781     36  5511.562     5513.872     5694.503\n",
      "WARNING: Number of threads seems too high for short alignments. Use -T AUTO to determine best number of threads.\n",
      "  3  GTR+F+G4      2720.070     36  5512.141     5514.451     5695.082\n",
      "WARNING: Number of threads seems too high for short alignments. Use -T AUTO to determine best number of threads.\n",
      "  4  GTR+F+I+G4    2719.793     37  5513.586     5516.027     5701.610\n",
      "WARNING: Number of threads seems too high for short alignments. Use -T AUTO to determine best number of threads.\n",
      "  5  GTR+F+R2      2720.160     37  5514.320     5516.761     5702.343\n",
      "WARNING: Number of threads seems too high for short alignments. Use -T AUTO to determine best number of threads.\n",
      "  6  GTR+F+R3      2720.042     39  5518.084     5520.797     5716.271\n",
      "WARNING: Number of threads seems too high for short alignments. Use -T AUTO to determine best number of threads.\n",
      " 14  SYM           2733.051     32  5530.102     5531.927     5692.716\n",
      "WARNING: Number of threads seems too high for short alignments. Use -T AUTO to determine best number of threads.\n",
      " 15  SYM+I         2730.478     33  5526.957     5528.898     5694.653\n",
      "WARNING: Number of threads seems too high for short alignments. Use -T AUTO to determine best number of threads.\n",
      " 16  SYM+G4        2730.757     33  5527.514     5529.455     5695.211\n",
      "WARNING: Number of threads seems too high for short alignments. Use -T AUTO to determine best number of threads.\n",
      " 17  SYM+I+G4      2730.494     34  5528.988     5531.049     5701.766\n",
      "WARNING: Number of threads seems too high for short alignments. Use -T AUTO to determine best number of threads.\n",
      " 18  SYM+R2        2730.842     34  5529.684     5531.745     5702.462\n",
      "WARNING: Number of threads seems too high for short alignments. Use -T AUTO to determine best number of threads.\n",
      " 27  TVM+F         2727.850     34  5523.701     5525.761     5696.479\n",
      "WARNING: Number of threads seems too high for short alignments. Use -T AUTO to determine best number of threads.\n",
      " 28  TVM+F+I       2725.093     35  5520.186     5522.369     5698.045\n",
      "WARNING: Number of threads seems too high for short alignments. Use -T AUTO to determine best number of threads.\n",
      " 29  TVM+F+G4      2725.467     35  5520.934     5523.117     5698.793\n",
      "WARNING: Number of threads seems too high for short alignments. Use -T AUTO to determine best number of threads.\n",
      " 30  TVM+F+I+G4    2725.098     36  5522.197     5524.507     5705.138\n",
      "WARNING: Number of threads seems too high for short alignments. Use -T AUTO to determine best number of threads.\n",
      " 31  TVM+F+R2      2725.531     36  5523.062     5525.373     5706.004\n",
      "WARNING: Number of threads seems too high for short alignments. Use -T AUTO to determine best number of threads.\n",
      " 40  TVMe          2740.551     31  5543.103     5544.816     5700.636\n",
      "WARNING: Number of threads seems too high for short alignments. Use -T AUTO to determine best number of threads.\n",
      " 41  TVMe+I        2737.661     32  5539.322     5541.147     5701.937\n",
      "WARNING: Number of threads seems too high for short alignments. Use -T AUTO to determine best number of threads.\n",
      " 42  TVMe+G4       2738.060     32  5540.120     5541.945     5702.735\n",
      "WARNING: Number of threads seems too high for short alignments. Use -T AUTO to determine best number of threads.\n",
      " 43  TVMe+I+G4     2737.663     33  5541.326     5543.268     5709.023\n",
      "WARNING: Number of threads seems too high for short alignments. Use -T AUTO to determine best number of threads.\n",
      " 44  TVMe+R2       2738.142     33  5542.284     5544.225     5709.980\n",
      "WARNING: Number of threads seems too high for short alignments. Use -T AUTO to determine best number of threads.\n",
      " 53  TIM3+F        2725.037     33  5516.074     5518.015     5683.770\n",
      "WARNING: Number of threads seems too high for short alignments. Use -T AUTO to determine best number of threads.\n",
      " 54  TIM3+F+I      2722.283     34  5512.566     5514.626     5685.344\n",
      "WARNING: Number of threads seems too high for short alignments. Use -T AUTO to determine best number of threads.\n",
      " 55  TIM3+F+G4     2722.599     34  5513.198     5515.259     5685.976\n",
      "WARNING: Number of threads seems too high for short alignments. Use -T AUTO to determine best number of threads.\n",
      " 56  TIM3+F+I+G4   2722.282     35  5514.564     5516.748     5692.424\n",
      "WARNING: Number of threads seems too high for short alignments. Use -T AUTO to determine best number of threads.\n",
      " 57  TIM3+F+R2     2722.752     35  5515.504     5517.688     5693.364\n",
      "WARNING: Number of threads seems too high for short alignments. Use -T AUTO to determine best number of threads.\n",
      " 66  TIM3e         2736.340     30  5532.680     5534.285     5685.131\n",
      "WARNING: Number of threads seems too high for short alignments. Use -T AUTO to determine best number of threads.\n",
      " 67  TIM3e+I       2733.555     31  5529.111     5530.824     5686.644\n",
      "WARNING: Number of threads seems too high for short alignments. Use -T AUTO to determine best number of threads.\n",
      " 68  TIM3e+G4      2733.879     31  5529.758     5531.472     5687.291\n",
      "WARNING: Number of threads seems too high for short alignments. Use -T AUTO to determine best number of threads.\n",
      " 69  TIM3e+I+G4    2733.558     32  5531.116     5532.941     5693.731\n",
      "WARNING: Number of threads seems too high for short alignments. Use -T AUTO to determine best number of threads.\n",
      " 70  TIM3e+R2      2734.010     32  5532.021     5533.846     5694.635\n",
      "WARNING: Number of threads seems too high for short alignments. Use -T AUTO to determine best number of threads.\n",
      " 79  TIM2+F        2722.594     33  5511.188     5513.129     5678.884\n",
      "WARNING: Number of threads seems too high for short alignments. Use -T AUTO to determine best number of threads.\n",
      " 80  TIM2+F+I      2720.061     34  5508.123     5510.183     5680.901\n",
      "WARNING: Number of threads seems too high for short alignments. Use -T AUTO to determine best number of threads.\n",
      " 81  TIM2+F+G4     2720.347     34  5508.695     5510.755     5681.473\n",
      "WARNING: Number of threads seems too high for short alignments. Use -T AUTO to determine best number of threads.\n",
      " 82  TIM2+F+I+G4   2720.073     35  5510.145     5512.329     5688.005\n",
      "WARNING: Number of threads seems too high for short alignments. Use -T AUTO to determine best number of threads.\n",
      " 83  TIM2+F+R2     2720.442     35  5510.884     5513.067     5688.744\n",
      "WARNING: Number of threads seems too high for short alignments. Use -T AUTO to determine best number of threads.\n",
      " 92  TIM2e         2733.680     30  5527.361     5528.966     5679.812\n",
      "WARNING: Number of threads seems too high for short alignments. Use -T AUTO to determine best number of threads.\n",
      " 93  TIM2e+I       2731.082     31  5524.164     5525.878     5681.697\n",
      "WARNING: Number of threads seems too high for short alignments. Use -T AUTO to determine best number of threads.\n",
      " 94  TIM2e+G4      2731.368     31  5524.735     5526.448     5682.268\n",
      "WARNING: Number of threads seems too high for short alignments. Use -T AUTO to determine best number of threads.\n",
      " 95  TIM2e+I+G4    2731.097     32  5526.194     5528.019     5688.808\n",
      "WARNING: Number of threads seems too high for short alignments. Use -T AUTO to determine best number of threads.\n",
      " 96  TIM2e+R2      2731.457     32  5526.913     5528.738     5689.528\n",
      "WARNING: Number of threads seems too high for short alignments. Use -T AUTO to determine best number of threads.\n",
      "105  TIM+F         2725.178     33  5516.356     5518.298     5684.053\n",
      "WARNING: Number of threads seems too high for short alignments. Use -T AUTO to determine best number of threads.\n",
      "106  TIM+F+I       2722.373     34  5512.745     5514.806     5685.523\n",
      "WARNING: Number of threads seems too high for short alignments. Use -T AUTO to determine best number of threads.\n",
      "107  TIM+F+G4      2722.694     34  5513.388     5515.449     5686.166\n",
      "WARNING: Number of threads seems too high for short alignments. Use -T AUTO to determine best number of threads.\n",
      "108  TIM+F+I+G4    2722.371     35  5514.742     5516.925     5692.601\n",
      "WARNING: Number of threads seems too high for short alignments. Use -T AUTO to determine best number of threads.\n",
      "109  TIM+F+R2      2722.862     35  5515.724     5517.908     5693.584\n",
      "WARNING: Number of threads seems too high for short alignments. Use -T AUTO to determine best number of threads.\n",
      "118  TIMe          2736.826     30  5533.652     5535.256     5686.103\n",
      "WARNING: Number of threads seems too high for short alignments. Use -T AUTO to determine best number of threads.\n",
      "119  TIMe+I        2733.988     31  5529.977     5531.690     5687.510\n",
      "WARNING: Number of threads seems too high for short alignments. Use -T AUTO to determine best number of threads.\n",
      "120  TIMe+G4       2734.321     31  5530.642     5532.355     5688.175\n",
      "WARNING: Number of threads seems too high for short alignments. Use -T AUTO to determine best number of threads.\n",
      "121  TIMe+I+G4     2733.990     32  5531.980     5533.806     5694.595\n",
      "WARNING: Number of threads seems too high for short alignments. Use -T AUTO to determine best number of threads.\n",
      "122  TIMe+R2       2734.466     32  5532.932     5534.758     5695.547\n",
      "WARNING: Number of threads seems too high for short alignments. Use -T AUTO to determine best number of threads.\n",
      "131  TPM3u+F       2730.603     32  5525.207     5527.032     5687.821\n",
      "WARNING: Number of threads seems too high for short alignments. Use -T AUTO to determine best number of threads.\n",
      "132  TPM3u+F+I     2727.678     33  5521.357     5523.298     5689.053\n",
      "WARNING: Number of threads seems too high for short alignments. Use -T AUTO to determine best number of threads.\n",
      "133  TPM3u+F+G4    2728.062     33  5522.124     5524.066     5689.821\n",
      "WARNING: Number of threads seems too high for short alignments. Use -T AUTO to determine best number of threads.\n",
      "134  TPM3u+F+I+G4  2727.679     34  5523.359     5525.420     5696.137\n",
      "WARNING: Number of threads seems too high for short alignments. Use -T AUTO to determine best number of threads.\n",
      "135  TPM3u+F+R2    2728.170     34  5524.340     5526.401     5697.119\n",
      "WARNING: Number of threads seems too high for short alignments. Use -T AUTO to determine best number of threads.\n",
      "144  TPM3+F        2730.603     32  5525.207     5527.032     5687.821\n",
      "WARNING: Number of threads seems too high for short alignments. Use -T AUTO to determine best number of threads.\n",
      "145  TPM3+F+I      2727.678     33  5521.357     5523.298     5689.053\n",
      "WARNING: Number of threads seems too high for short alignments. Use -T AUTO to determine best number of threads.\n",
      "146  TPM3+F+G4     2728.062     33  5522.124     5524.066     5689.821\n",
      "WARNING: Number of threads seems too high for short alignments. Use -T AUTO to determine best number of threads.\n",
      "147  TPM3+F+I+G4   2727.679     34  5523.359     5525.420     5696.137\n",
      "WARNING: Number of threads seems too high for short alignments. Use -T AUTO to determine best number of threads.\n",
      "148  TPM3+F+R2     2728.170     34  5524.340     5526.401     5697.119\n",
      "WARNING: Number of threads seems too high for short alignments. Use -T AUTO to determine best number of threads.\n",
      "157  TPM2u+F       2728.127     32  5520.254     5522.080     5682.869\n",
      "WARNING: Number of threads seems too high for short alignments. Use -T AUTO to determine best number of threads.\n",
      "158  TPM2u+F+I     2725.359     33  5516.719     5518.660     5684.415\n",
      "WARNING: Number of threads seems too high for short alignments. Use -T AUTO to determine best number of threads.\n",
      "159  TPM2u+F+G4    2725.726     33  5517.452     5519.393     5685.149\n",
      "WARNING: Number of threads seems too high for short alignments. Use -T AUTO to determine best number of threads.\n",
      "160  TPM2u+F+I+G4  2725.364     34  5518.729     5520.789     5691.507\n",
      "WARNING: Number of threads seems too high for short alignments. Use -T AUTO to determine best number of threads.\n",
      "161  TPM2u+F+R2    2725.789     34  5519.579     5521.639     5692.357\n",
      "WARNING: Number of threads seems too high for short alignments. Use -T AUTO to determine best number of threads.\n",
      "170  TPM2+F        2728.127     32  5520.254     5522.080     5682.869\n",
      "WARNING: Number of threads seems too high for short alignments. Use -T AUTO to determine best number of threads.\n",
      "171  TPM2+F+I      2725.359     33  5516.719     5518.660     5684.415\n",
      "WARNING: Number of threads seems too high for short alignments. Use -T AUTO to determine best number of threads.\n",
      "172  TPM2+F+G4     2725.726     33  5517.452     5519.393     5685.149\n",
      "WARNING: Number of threads seems too high for short alignments. Use -T AUTO to determine best number of threads.\n",
      "173  TPM2+F+I+G4   2725.364     34  5518.729     5520.789     5691.507\n",
      "WARNING: Number of threads seems too high for short alignments. Use -T AUTO to determine best number of threads.\n",
      "174  TPM2+F+R2     2725.789     34  5519.579     5521.639     5692.357\n",
      "WARNING: Number of threads seems too high for short alignments. Use -T AUTO to determine best number of threads.\n",
      "183  K3Pu+F        2730.726     32  5525.452     5527.277     5688.066\n",
      "WARNING: Number of threads seems too high for short alignments. Use -T AUTO to determine best number of threads.\n",
      "184  K3Pu+F+I      2727.747     33  5521.495     5523.436     5689.191\n",
      "WARNING: Number of threads seems too high for short alignments. Use -T AUTO to determine best number of threads.\n",
      "185  K3Pu+F+G4     2728.137     33  5522.274     5524.215     5689.970\n",
      "WARNING: Number of threads seems too high for short alignments. Use -T AUTO to determine best number of threads.\n",
      "186  K3Pu+F+I+G4   2727.748     34  5523.496     5525.556     5696.274\n",
      "WARNING: Number of threads seems too high for short alignments. Use -T AUTO to determine best number of threads.\n",
      "187  K3Pu+F+R2     2728.260     34  5524.521     5526.582     5697.299\n",
      "WARNING: Number of threads seems too high for short alignments. Use -T AUTO to determine best number of threads.\n",
      "196  K3P           2744.329     29  5546.658     5548.158     5694.028\n",
      "WARNING: Number of threads seems too high for short alignments. Use -T AUTO to determine best number of threads.\n",
      "197  K3P+I         2741.196     30  5542.392     5543.997     5694.844\n",
      "WARNING: Number of threads seems too high for short alignments. Use -T AUTO to determine best number of threads.\n",
      "198  K3P+G4        2741.637     30  5543.273     5544.878     5695.725\n",
      "WARNING: Number of threads seems too high for short alignments. Use -T AUTO to determine best number of threads.\n",
      "199  K3P+I+G4      2741.200     31  5544.401     5546.114     5701.934\n",
      "WARNING: Number of threads seems too high for short alignments. Use -T AUTO to determine best number of threads.\n",
      "200  K3P+R2        2741.786     31  5545.572     5547.285     5703.105\n",
      "WARNING: Number of threads seems too high for short alignments. Use -T AUTO to determine best number of threads.\n",
      "209  TN+F          2725.177     32  5514.354     5516.179     5676.968\n",
      "WARNING: Number of threads seems too high for short alignments. Use -T AUTO to determine best number of threads.\n",
      "210  TN+F+I        2722.392     33  5510.784     5512.725     5678.480\n",
      "WARNING: Number of threads seems too high for short alignments. Use -T AUTO to determine best number of threads.\n",
      "211  TN+F+G4       2722.709     33  5511.419     5513.360     5679.115\n",
      "WARNING: Number of threads seems too high for short alignments. Use -T AUTO to determine best number of threads.\n",
      "212  TN+F+I+G4     2722.391     34  5512.783     5514.844     5685.561\n",
      "WARNING: Number of threads seems too high for short alignments. Use -T AUTO to determine best number of threads.\n",
      "213  TN+F+R2       2722.870     34  5513.740     5515.801     5686.518\n",
      "WARNING: Number of threads seems too high for short alignments. Use -T AUTO to determine best number of threads.\n",
      "222  TNe           2736.825     29  5531.651     5533.151     5679.020\n",
      "WARNING: Number of threads seems too high for short alignments. Use -T AUTO to determine best number of threads.\n",
      "223  TNe+I         2734.000     30  5528.001     5529.606     5680.452\n",
      "WARNING: Number of threads seems too high for short alignments. Use -T AUTO to determine best number of threads.\n",
      "224  TNe+G4        2734.329     30  5528.658     5530.263     5681.110\n",
      "WARNING: Number of threads seems too high for short alignments. Use -T AUTO to determine best number of threads.\n",
      "225  TNe+I+G4      2734.000     31  5529.999     5531.712     5687.532\n",
      "WARNING: Number of threads seems too high for short alignments. Use -T AUTO to determine best number of threads.\n",
      "226  TNe+R2        2734.466     31  5530.933     5532.646     5688.466\n",
      "WARNING: Number of threads seems too high for short alignments. Use -T AUTO to determine best number of threads.\n",
      "235  HKY+F         2730.727     31  5523.454     5525.167     5680.987\n",
      "WARNING: Number of threads seems too high for short alignments. Use -T AUTO to determine best number of threads.\n",
      "236  HKY+F+I       2727.762     32  5519.525     5521.350     5682.140\n",
      "WARNING: Number of threads seems too high for short alignments. Use -T AUTO to determine best number of threads.\n",
      "237  HKY+F+G4      2728.149     32  5520.298     5522.123     5682.913\n",
      "WARNING: Number of threads seems too high for short alignments. Use -T AUTO to determine best number of threads.\n",
      "238  HKY+F+I+G4    2727.763     33  5521.526     5523.467     5689.222\n",
      "WARNING: Number of threads seems too high for short alignments. Use -T AUTO to determine best number of threads.\n",
      "239  HKY+F+R2      2728.267     33  5522.534     5524.475     5690.230\n",
      "WARNING: Number of threads seems too high for short alignments. Use -T AUTO to determine best number of threads.\n",
      "248  K2P           2744.329     28  5544.658     5546.057     5686.946\n",
      "WARNING: Number of threads seems too high for short alignments. Use -T AUTO to determine best number of threads.\n",
      "249  K2P+I         2741.207     29  5540.415     5541.915     5687.784\n",
      "WARNING: Number of threads seems too high for short alignments. Use -T AUTO to determine best number of threads.\n",
      "250  K2P+G4        2741.643     29  5541.287     5542.787     5688.656\n",
      "WARNING: Number of threads seems too high for short alignments. Use -T AUTO to determine best number of threads.\n",
      "251  K2P+I+G4      2741.212     30  5542.424     5544.029     5694.875\n",
      "WARNING: Number of threads seems too high for short alignments. Use -T AUTO to determine best number of threads.\n",
      "252  K2P+R2        2741.788     30  5543.576     5545.181     5696.027\n",
      "WARNING: Number of threads seems too high for short alignments. Use -T AUTO to determine best number of threads.\n",
      "261  F81+F         2807.910     30  5675.820     5677.424     5828.271\n",
      "WARNING: Number of threads seems too high for short alignments. Use -T AUTO to determine best number of threads.\n",
      "262  F81+F+I       2806.548     31  5675.096     5676.809     5832.629\n",
      "WARNING: Number of threads seems too high for short alignments. Use -T AUTO to determine best number of threads.\n",
      "263  F81+F+G4      2806.627     31  5675.253     5676.967     5832.786\n",
      "WARNING: Number of threads seems too high for short alignments. Use -T AUTO to determine best number of threads.\n",
      "264  F81+F+I+G4    2806.714     32  5677.428     5679.254     5840.043\n",
      "WARNING: Number of threads seems too high for short alignments. Use -T AUTO to determine best number of threads.\n",
      "265  F81+F+R2      2806.601     32  5677.203     5679.028     5839.817\n",
      "WARNING: Number of threads seems too high for short alignments. Use -T AUTO to determine best number of threads.\n",
      "274  JC            2820.675     27  5695.350     5696.652     5832.557\n",
      "WARNING: Number of threads seems too high for short alignments. Use -T AUTO to determine best number of threads.\n",
      "275  JC+I          2819.209     28  5694.419     5695.817     5836.706\n",
      "WARNING: Number of threads seems too high for short alignments. Use -T AUTO to determine best number of threads.\n",
      "276  JC+G4         2819.298     28  5694.597     5695.995     5836.884\n",
      "WARNING: Number of threads seems too high for short alignments. Use -T AUTO to determine best number of threads.\n",
      "277  JC+I+G4       2819.332     29  5696.665     5698.165     5844.034\n",
      "WARNING: Number of threads seems too high for short alignments. Use -T AUTO to determine best number of threads.\n",
      "278  JC+R2         2819.280     29  5696.560     5698.060     5843.930\n",
      "Akaike Information Criterion:           TIM2+F+I\n",
      "Corrected Akaike Information Criterion: TIM2+F+I\n",
      "Bayesian Information Criterion:         TN+F\n",
      "Best-fit model: TN+F chosen according to BIC\n",
      "\n",
      "All model information printed to output/FeLV-B.mafft.selectB1.aln.fasta.model.gz\n",
      "CPU time for ModelFinder: 60.776 seconds (0h:1m:0s)\n",
      "Wall-clock time for ModelFinder: 31.447 seconds (0h:0m:31s)\n",
      "Generating 1000 samples for ultrafast bootstrap (seed: 337002)...\n",
      "\n",
      "NOTE: 0 MB RAM (0 GB) is required!\n",
      "WARNING: Number of threads seems too high for short alignments. Use -T AUTO to determine best number of threads.\n",
      "Estimate model parameters (epsilon = 0.100)\n",
      "1. Initial log-likelihood: -2725.177\n",
      "Optimal log-likelihood: -2725.177\n",
      "Rate parameters:  A-C: 1.00000  A-G: 7.10741  A-T: 1.00000  C-G: 1.00000  C-T: 4.07693  G-T: 1.00000\n",
      "Base frequencies:  A: 0.279  C: 0.267  G: 0.241  T: 0.213\n",
      "Parameters optimization took 1 rounds (0.000 sec)\n",
      "Computing ML distances based on estimated model parameters...\n",
      "Computing ML distances took 0.000579 sec (of wall-clock time) 0.001140 sec(of CPU time)\n",
      "Computing RapidNJ tree took 0.000107 sec (of wall-clock time) 0.000176 sec (of CPU time)\n",
      "Log-likelihood of RapidNJ tree: -2855.647\n",
      "--------------------------------------------------------------------\n",
      "|             INITIALIZING CANDIDATE TREE SET                      |\n",
      "--------------------------------------------------------------------\n",
      "Generating 98 parsimony trees... 0.036 second\n",
      "Computing log-likelihood of 98 initial trees ... 0.027 seconds\n",
      "Current best score: -2725.177\n",
      "\n",
      "Do NNI search on 20 best initial trees\n",
      "Estimate model parameters (epsilon = 0.100)\n",
      "BETTER TREE FOUND at iteration 1: -2725.175\n",
      "BETTER TREE FOUND at iteration 2: -2725.174\n",
      "UPDATE BEST LOG-LIKELIHOOD: -2725.174\n",
      "UPDATE BEST LOG-LIKELIHOOD: -2725.174\n",
      "Iteration 10 / LogL: -2725.175 / Time: 0h:0m:0s\n",
      "Iteration 20 / LogL: -2725.185 / Time: 0h:0m:0s\n",
      "Finish initializing candidate tree set (5)\n",
      "Current best tree score: -2725.174 / CPU time: 0.111\n",
      "Number of iterations: 20\n",
      "--------------------------------------------------------------------\n",
      "|               OPTIMIZING CANDIDATE TREE SET                      |\n",
      "--------------------------------------------------------------------\n",
      "Iteration 30 / LogL: -2725.227 / Time: 0h:0m:0s (0h:0m:0s left)\n",
      "Iteration 40 / LogL: -2725.185 / Time: 0h:0m:0s (0h:0m:0s left)\n",
      "Iteration 50 / LogL: -2725.232 / Time: 0h:0m:0s (0h:0m:0s left)\n",
      "Log-likelihood cutoff on original alignment: -2747.867\n",
      "Iteration 60 / LogL: -2725.196 / Time: 0h:0m:0s (0h:0m:0s left)\n",
      "Iteration 70 / LogL: -2725.189 / Time: 0h:0m:0s (0h:0m:0s left)\n",
      "Iteration 80 / LogL: -2725.177 / Time: 0h:0m:0s (0h:0m:0s left)\n",
      "Iteration 90 / LogL: -2725.174 / Time: 0h:0m:0s (0h:0m:0s left)\n",
      "Iteration 100 / LogL: -2725.192 / Time: 0h:0m:0s (0h:0m:0s left)\n",
      "Log-likelihood cutoff on original alignment: -2749.519\n",
      "NOTE: Bootstrap correlation coefficient of split occurrence frequencies: 0.994\n",
      "UPDATE BEST LOG-LIKELIHOOD: -2725.174\n",
      "TREE SEARCH COMPLETED AFTER 103 ITERATIONS / Time: 0h:0m:0s\n",
      "\n",
      "--------------------------------------------------------------------\n",
      "|                    FINALIZING TREE SEARCH                        |\n",
      "--------------------------------------------------------------------\n",
      "Performs final model parameters optimization\n",
      "Estimate model parameters (epsilon = 0.010)\n",
      "1. Initial log-likelihood: -2725.174\n",
      "Optimal log-likelihood: -2725.174\n",
      "Rate parameters:  A-C: 1.00000  A-G: 7.10656  A-T: 1.00000  C-G: 1.00000  C-T: 4.07694  G-T: 1.00000\n",
      "Base frequencies:  A: 0.279  C: 0.267  G: 0.241  T: 0.213\n",
      "Parameters optimization took 1 rounds (0.000 sec)\n",
      "BEST SCORE FOUND : -2725.174\n",
      "Creating bootstrap support values...\n",
      "Split supports printed to NEXUS file output/FeLV-B.mafft.selectB1.aln.fasta.splits.nex\n",
      "Total tree length: 0.262\n",
      "\n",
      "Total number of iterations: 103\n",
      "CPU time used for tree search: 1.562 sec (0h:0m:1s)\n",
      "Wall-clock time used for tree search: 0.708 sec (0h:0m:0s)\n",
      "Total CPU time used: 1.660 sec (0h:0m:1s)\n",
      "Total wall-clock time used: 0.760 sec (0h:0m:0s)\n",
      "\n",
      "Computing bootstrap consensus tree...\n",
      "Reading input file output/FeLV-B.mafft.selectB1.aln.fasta.splits.nex...\n",
      "15 taxa and 81 splits.\n",
      "Consensus tree written to output/FeLV-B.mafft.selectB1.aln.fasta.contree\n",
      "Reading input trees file output/FeLV-B.mafft.selectB1.aln.fasta.contree\n",
      "Log-likelihood of consensus tree: -2725.192\n",
      "\n",
      "Analysis results written to: \n",
      "  IQ-TREE report:                output/FeLV-B.mafft.selectB1.aln.fasta.iqtree\n",
      "  Maximum-likelihood tree:       output/FeLV-B.mafft.selectB1.aln.fasta.treefile\n",
      "  Likelihood distances:          output/FeLV-B.mafft.selectB1.aln.fasta.mldist\n",
      "\n",
      "Ultrafast bootstrap approximation results written to:\n",
      "  Split support values:          output/FeLV-B.mafft.selectB1.aln.fasta.splits.nex\n",
      "  Consensus tree:                output/FeLV-B.mafft.selectB1.aln.fasta.contree\n",
      "  Screen log file:               output/FeLV-B.mafft.selectB1.aln.fasta.log\n",
      "\n",
      "Date and Time: Wed Dec 20 10:42:43 2023\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "[INFO]\u001b[0m read sequences ...\n",
      "[INFO]\u001b[0m 15 sequences loaded\n",
      "[INFO]\u001b[0m sorting ...\n",
      "[INFO]\u001b[0m output ...\n",
      "outputhat23=16\n",
      "treein = 0\n",
      "compacttree = 0\n",
      "stacksize: 8192 kb\n",
      "generating a scoring matrix for nucleotide (dist=200) ... done\n",
      "All-to-all alignment.\n",
      "    0 / 15\r",
      "    1 / 15\r",
      "    2 / 15\r",
      "    3 / 15\r",
      "    4 / 15\r",
      "    5 / 15\r",
      "    6 / 15\r",
      "    7 / 15\r",
      "    8 / 15\r",
      "    9 / 15\r",
      "   10 / 15\r",
      "   11 / 15\r",
      "   12 / 15\r",
      "   13 / 15\r",
      "tbfast-pair (nuc) Version 7.407\n",
      "alg=L, model=DNA200 (2), 2.00 (6.00), -0.10 (-0.30), noshift, amax=0.0\n",
      "0 thread(s)\n",
      "\n",
      "outputhat23=16\n",
      "Loading 'hat3.seed' ... \n",
      "done.\n",
      "Writing hat3 for iterative refinement\n",
      "generating a scoring matrix for nucleotide (dist=200) ... done\n",
      "Gap Penalty = -1.53, +0.00, +0.00\n",
      "tbutree = 1, compacttree = 0\n",
      "Constructing a UPGMA tree ... \n",
      "\r",
      "    0 / 15\r",
      "   10 / 15\n",
      "done.\n",
      "\n",
      "Progressive alignment ... \n",
      "\r",
      "STEP     1 /14 \r",
      "STEP     2 /14 \r",
      "STEP     3 /14 \r",
      "STEP     4 /14 \r",
      "STEP     5 /14 \r",
      "STEP     6 /14 \r",
      "STEP     7 /14 \r",
      "STEP     8 /14 \r",
      "STEP     9 /14 \r",
      "STEP    10 /14 \r",
      "STEP    11 /14 \r",
      "STEP    12 /14 \r",
      "STEP    13 /14 \r",
      "STEP    14 /14 \n",
      "done.\n",
      "tbfast (nuc) Version 7.407\n",
      "alg=A, model=DNA200 (2), 1.53 (4.59), -0.00 (-0.00), noshift, amax=0.0\n",
      "1 thread(s)\n",
      "\n",
      "minimumweight = 0.000010\n",
      "autosubalignment = 0.000000\n",
      "nthread = 0\n",
      "randomseed = 0\n",
      "blosum 62 / kimura 200\n",
      "poffset = 0\n",
      "niter = 16\n",
      "sueff_global = 0.100000\n",
      "nadd = 16\n",
      "Loading 'hat3' ... done.\n",
      "generating a scoring matrix for nucleotide (dist=200) ... done\n",
      "\n",
      "\r",
      "    0 / 15\r",
      "   10 / 15\n",
      "Segment   1/  1    1-1191\n",
      "STEP 001-001-0  identical.   \r",
      "STEP 001-001-1  identical.   \r",
      "STEP 001-002-0  identical.   \r",
      "STEP 001-002-1  identical.   \r",
      "STEP 001-003-0  identical.   \r",
      "STEP 001-003-1  identical.   \r",
      "STEP 001-004-0  identical.   \r",
      "STEP 001-004-1  identical.   \r",
      "STEP 001-005-0  identical.   \r",
      "STEP 001-005-1  identical.   \r",
      "STEP 001-006-0  identical.   \r",
      "STEP 001-006-1  identical.   \r",
      "STEP 001-007-0  identical.   \r",
      "STEP 001-007-1  identical.   \r",
      "STEP 001-008-0  identical.   \r",
      "STEP 001-008-1  identical.   \r",
      "STEP 001-009-0  identical.   \r",
      "STEP 001-009-1  identical.   \r",
      "STEP 001-010-0  identical.   \r",
      "STEP 001-010-1  identical.   \r",
      "STEP 001-011-0  identical.   \r",
      "STEP 001-011-1  identical.   \r",
      "STEP 001-012-0  identical.   \r",
      "STEP 001-012-1  identical.   \r",
      "STEP 001-013-0  identical.   \r",
      "STEP 001-013-1  identical.   \r",
      "STEP 001-014-1  identical.   \r",
      "STEP 002-014-1  identical.   \r",
      "STEP 002-013-0  identical.   \r",
      "STEP 002-013-1  identical.   \r\n",
      "Converged.\n",
      "\n",
      "done\n",
      "dvtditr (nuc) Version 7.407\n",
      "alg=A, model=DNA200 (2), 1.53 (4.59), -0.00 (-0.00), noshift, amax=0.0\n",
      "0 thread(s)\n",
      "\n",
      "\n",
      "Strategy:\n",
      " L-INS-i (Probably most accurate, very slow)\n",
      " Iterative refinement method (<16) with LOCAL pairwise alignment information\n",
      "\n",
      "If unsure which option to use, try 'mafft --auto input > output'.\n",
      "For more information, see 'mafft --help', 'mafft --man' and the mafft page.\n",
      "\n",
      "The default gap scoring scheme has been changed in version 7.110 (2013 Oct).\n",
      "It tends to insert more gaps into gap-rich regions than previous versions.\n",
      "To disable this change, add the --leavegappyregion option.\n",
      "\n",
      "OMP: Info #273: omp_set_nested routine deprecated, please use omp_set_max_active_levels instead.\n",
      "OMP: Info #273: omp_set_nested routine deprecated, please use omp_set_max_active_levels instead.\n",
      "OMP: Info #273: omp_set_nested routine deprecated, please use omp_set_max_active_levels instead.\n"
     ]
    }
   ],
   "source": [
    "%%bash\n",
    "\n",
    "cat output/FeLV-B.mafft.smplot.fasta | \\\n",
    "    seqkit subseq -r 1:1190 | \\\n",
    "    seqkit grep -n -r -p \"AB674444\" -v | \\\n",
    "    seqkit sort --by-name \\\n",
    "    > output/FeLV-B.mafft.selectB1.fasta\n",
    "    \n",
    "mafft \\\n",
    "    --auto \\\n",
    "    output/FeLV-B.mafft.selectB1.fasta \\\n",
    "    > output/FeLV-B.mafft.selectB1.aln.fasta\n",
    "    \n",
    "iqtree \\\n",
    "    -s output/FeLV-B.mafft.selectB1.aln.fasta \\\n",
    "    -m TESTNEW \\\n",
    "    -redo \\\n",
    "    -bb 1000 \\\n",
    "    -nt AUTO"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# Phylogenetic analysis for the FeLV-A derived region"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## site 1619-2550"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 13,
   "metadata": {
    "ExecuteTime": {
     "end_time": "2023-12-20T01:43:18.607387Z",
     "start_time": "2023-12-20T01:42:43.561007Z"
    },
    "run_control": {
     "marked": true
    },
    "scrolled": true
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "IQ-TREE multicore version 2.1.4-beta COVID-edition for Mac OS X 64-bit built Jun 24 2021\n",
      "Developed by Bui Quang Minh, James Barbetti, Nguyen Lam Tung,\n",
      "Olga Chernomor, Heiko Schmidt, Dominik Schrempf, Michael Woodhams.\n",
      "\n",
      "Host:    iMac.local (AVX2, FMA3, 64 GB RAM)\n",
      "Command: iqtree -s output/FeLV-B.mafft.selectA1.aln.fasta -m TESTNEW -redo -bb 1000 -nt AUTO\n",
      "Seed:    662094 (Using SPRNG - Scalable Parallel Random Number Generator)\n",
      "Time:    Wed Dec 20 10:42:45 2023\n",
      "Kernel:  AVX+FMA - auto-detect threads (12 CPU cores detected)\n",
      "\n",
      "Reading alignment file output/FeLV-B.mafft.selectA1.aln.fasta ... Fasta format detected\n",
      "Alignment most likely contains DNA/RNA sequences\n",
      "Alignment has 24 sequences with 932 columns, 96 distinct patterns\n",
      "114 parsimony-informative, 22 singleton sites, 796 constant sites\n",
      "                        Gap/Ambiguity  Composition  p-value\n",
      "   1  Y_X00188.1                2.15%    passed     92.37%\n",
      "   2  FeLV-B_ON-T_clone_9      29.61%    passed     98.47%\n",
      "   3  FeLV-B_ON-T_clone_3.6    30.79%    passed     83.00%\n",
      "   4  FeLV-B_ON-T_clone_3.14   30.79%    passed     83.00%\n",
      "   5  FeLV-B_ON-T_clone_11     29.61%    passed     99.29%\n",
      "   6  FeLV-B_ON-T__clone_2     30.79%    passed     99.33%\n",
      "   7  FeLV-B_ON-T__clone_14    30.79%    passed     98.62%\n",
      "   8  FeLV-B_ON-H__clone16     30.79%    passed     99.33%\n",
      "   9  FeLV-B_ON-C_clone_9.14    0.32%    passed     86.81%\n",
      "  10  FeLV-B_ON-C_clone_9.1     0.00%    passed     89.36%\n",
      "  11  FeLV-B_ON-C_clone_13     30.79%    passed     99.33%\n",
      "  12  FeLV-B_ON-C__clone_14    30.79%    passed     99.03%\n",
      "  13  FeLV-B_ON-C__clone_1     30.79%    passed     99.33%\n",
      "  14  A_M18247.1                2.15%    passed     81.49%\n",
      "  15  FeLV-A_ON-H__clone_1     30.79%    passed     99.33%\n",
      "  16  FeLV-A_ON-H__clone_2     30.79%    passed     99.33%\n",
      "  17  FeLV-A_ON-T_clone_3.2     1.93%    passed     87.27%\n",
      "  18  FeLV-A_ON-T_clone_1.1    29.72%    passed     99.48%\n",
      "  19  FeLV-A_ON-C_clone_9.2     6.55%    passed     96.29%\n",
      "  20  FeLV-A_ON-T_clone_1.2    29.72%    passed     99.48%\n",
      "  21  FeLV-A_ON-C__clone_3     30.79%    passed     99.33%\n",
      "  22  FeLV-A_ON-T_clone_3.4     1.93%    passed     89.68%\n",
      "  23  FeLV-A_ON-C_clone_9.4     6.55%    passed     96.29%\n",
      "  24  FeLV-A_ON-C_clone_4      30.79%    passed     99.33%\n",
      "****  TOTAL                    21.24%  0 sequences failed composition chi2 test (p-value<5%; df=3)\n",
      "NOTE: FeLV-B_ON-H__clone16 is identical to FeLV-B_ON-T__clone_2 but kept for subsequent analysis\n",
      "NOTE: FeLV-A_ON-T_clone_1.2 is identical to FeLV-A_ON-T_clone_1.1 but kept for subsequent analysis\n",
      "NOTE: FeLV-A_ON-C_clone_9.4 is identical to FeLV-A_ON-C_clone_9.2 but kept for subsequent analysis\n",
      "NOTE: 6 identical sequences (see below) will be ignored for subsequent analysis\n",
      "NOTE: FeLV-B_ON-C_clone_13 (identical to FeLV-B_ON-T__clone_2) is ignored but added at the end\n",
      "NOTE: FeLV-B_ON-C__clone_1 (identical to FeLV-B_ON-T__clone_2) is ignored but added at the end\n",
      "NOTE: FeLV-A_ON-H__clone_1 (identical to FeLV-B_ON-T__clone_2) is ignored but added at the end\n",
      "NOTE: FeLV-A_ON-H__clone_2 (identical to FeLV-B_ON-T__clone_2) is ignored but added at the end\n",
      "NOTE: FeLV-A_ON-C__clone_3 (identical to FeLV-B_ON-T__clone_2) is ignored but added at the end\n",
      "NOTE: FeLV-A_ON-C_clone_4 (identical to FeLV-B_ON-T__clone_2) is ignored but added at the end\n",
      "Alignment was printed to output/FeLV-B.mafft.selectA1.aln.fasta.uniqueseq.phy\n",
      "\n",
      "For your convenience alignment with unique sequences printed to output/FeLV-B.mafft.selectA1.aln.fasta.uniqueseq.phy\n",
      "\n",
      "\n",
      "Create initial parsimony tree by phylogenetic likelihood library (PLL)... 0.000 seconds\n",
      "Measuring multi-threading efficiency up to 12 CPU cores\n",
      "Increase to 10 rounds for branch lengths\n",
      "8038 trees examined\n",
      "Threads: 1 / Time: 12.001 sec / Speedup: 1.000 / Efficiency: 100% / LogL: -2502\n",
      "Threads: 2 / Time: 9.562 sec / Speedup: 1.255 / Efficiency: 63% / LogL: -2502\n",
      "Threads: 3 / Time: 9.063 sec / Speedup: 1.324 / Efficiency: 44% / LogL: -2502\n",
      "BEST NUMBER OF THREADS: 2\n",
      "\n",
      "Perform fast likelihood tree search using GTR+I+G model...\n",
      "Estimate model parameters (epsilon = 5.000)\n",
      "Perform nearest neighbor interchange...\n",
      "Estimate model parameters (epsilon = 1.000)\n",
      "1. Initial log-likelihood: -2012.355\n",
      "Optimal log-likelihood: -2012.354\n",
      "Rate parameters:  A-C: 9.95836  A-G: 26.82855  A-T: 2.79227  C-G: 3.16460  C-T: 19.10998  G-T: 1.00000\n",
      "Base frequencies:  A: 0.314  C: 0.244  G: 0.206  T: 0.236\n",
      "Proportion of invariable sites: 0.428\n",
      "Gamma shape alpha: 3.566\n",
      "Parameters optimization took 1 rounds (0.001 sec)\n",
      "Time for fast ML tree search: 0.020 seconds\n",
      "\n",
      "NOTE: ModelFinder requires 0 MB RAM!\n",
      "ModelFinder will test up to 286 DNA models (sample size: 932) ...\n",
      " No. Model         -LnL         df  AIC          AICc         BIC\n",
      "WARNING: Number of threads seems too high for short alignments. Use -T AUTO to determine best number of threads.\n",
      "  1  GTR+F         2015.233     41  4112.467     4116.337     4310.798\n",
      "WARNING: Number of threads seems too high for short alignments. Use -T AUTO to determine best number of threads.\n",
      "  2  GTR+F+I       2012.314     42  4108.628     4112.691     4311.796\n",
      "WARNING: Number of threads seems too high for short alignments. Use -T AUTO to determine best number of threads.\n",
      "  3  GTR+F+G4      2012.398     42  4108.795     4112.858     4311.963\n",
      "WARNING: Number of threads seems too high for short alignments. Use -T AUTO to determine best number of threads.\n",
      "  4  GTR+F+I+G4    2012.363     43  4110.725     4114.987     4318.731\n",
      "WARNING: Number of threads seems too high for short alignments. Use -T AUTO to determine best number of threads.\n",
      "  5  GTR+F+R2      2013.146     43  4112.292     4116.553     4320.297\n",
      "WARNING: Number of threads seems too high for short alignments. Use -T AUTO to determine best number of threads.\n",
      "  6  GTR+F+R3      2012.680     45  4115.360     4120.033     4333.040\n",
      "WARNING: Number of threads seems too high for short alignments. Use -T AUTO to determine best number of threads.\n",
      " 14  SYM           2027.268     38  4130.536     4133.856     4314.355\n",
      "WARNING: Number of threads seems too high for short alignments. Use -T AUTO to determine best number of threads.\n",
      " 15  SYM+I         2024.526     39  4127.052     4130.550     4315.708\n",
      "WARNING: Number of threads seems too high for short alignments. Use -T AUTO to determine best number of threads.\n",
      " 16  SYM+G4        2024.547     39  4127.093     4130.591     4315.749\n",
      "WARNING: Number of threads seems too high for short alignments. Use -T AUTO to determine best number of threads.\n",
      " 17  SYM+I+G4      2024.535     40  4129.070     4132.751     4322.563\n",
      "WARNING: Number of threads seems too high for short alignments. Use -T AUTO to determine best number of threads.\n",
      " 18  SYM+R2        2025.228     40  4130.457     4134.138     4323.950\n",
      "WARNING: Number of threads seems too high for short alignments. Use -T AUTO to determine best number of threads.\n",
      " 27  TVM+F         2016.503     40  4113.006     4116.687     4306.499\n",
      "WARNING: Number of threads seems too high for short alignments. Use -T AUTO to determine best number of threads.\n",
      " 28  TVM+F+I       2013.380     41  4108.761     4112.630     4307.091\n",
      "WARNING: Number of threads seems too high for short alignments. Use -T AUTO to determine best number of threads.\n",
      " 29  TVM+F+G4      2013.488     41  4108.977     4112.846     4307.307\n",
      "WARNING: Number of threads seems too high for short alignments. Use -T AUTO to determine best number of threads.\n",
      " 30  TVM+F+I+G4    2013.455     42  4110.910     4114.973     4314.078\n",
      "WARNING: Number of threads seems too high for short alignments. Use -T AUTO to determine best number of threads.\n",
      " 31  TVM+F+R2      2014.348     42  4112.696     4116.759     4315.864\n",
      "WARNING: Number of threads seems too high for short alignments. Use -T AUTO to determine best number of threads.\n",
      " 40  TVMe          2028.854     37  4131.709     4134.854     4310.690\n",
      "WARNING: Number of threads seems too high for short alignments. Use -T AUTO to determine best number of threads.\n",
      " 41  TVMe+I        2025.888     38  4127.776     4131.095     4311.595\n",
      "WARNING: Number of threads seems too high for short alignments. Use -T AUTO to determine best number of threads.\n",
      " 42  TVMe+G4       2025.955     38  4127.909     4131.228     4311.728\n",
      "WARNING: Number of threads seems too high for short alignments. Use -T AUTO to determine best number of threads.\n",
      " 43  TVMe+I+G4     2025.925     39  4129.850     4133.348     4318.506\n",
      "WARNING: Number of threads seems too high for short alignments. Use -T AUTO to determine best number of threads.\n",
      " 44  TVMe+R2       2026.740     39  4131.480     4134.978     4320.136\n",
      "WARNING: Number of threads seems too high for short alignments. Use -T AUTO to determine best number of threads.\n",
      " 53  TIM3+F        2019.355     39  4116.710     4120.207     4305.366\n",
      "WARNING: Number of threads seems too high for short alignments. Use -T AUTO to determine best number of threads.\n",
      " 54  TIM3+F+I      2016.087     40  4112.175     4115.856     4305.668\n",
      "WARNING: Number of threads seems too high for short alignments. Use -T AUTO to determine best number of threads.\n",
      " 55  TIM3+F+G4     2016.175     40  4112.350     4116.031     4305.843\n",
      "WARNING: Number of threads seems too high for short alignments. Use -T AUTO to determine best number of threads.\n",
      " 56  TIM3+F+I+G4   2016.149     41  4114.297     4118.167     4312.628\n",
      "WARNING: Number of threads seems too high for short alignments. Use -T AUTO to determine best number of threads.\n",
      " 57  TIM3+F+R2     2017.125     41  4116.249     4120.119     4314.580\n",
      "WARNING: Number of threads seems too high for short alignments. Use -T AUTO to determine best number of threads.\n",
      " 66  TIM3e         2033.441     36  4138.881     4141.858     4313.025\n",
      "WARNING: Number of threads seems too high for short alignments. Use -T AUTO to determine best number of threads.\n",
      " 67  TIM3e+I       2030.577     37  4135.153     4138.299     4314.135\n",
      "WARNING: Number of threads seems too high for short alignments. Use -T AUTO to determine best number of threads.\n",
      " 68  TIM3e+G4      2030.627     37  4135.255     4138.400     4314.236\n",
      "WARNING: Number of threads seems too high for short alignments. Use -T AUTO to determine best number of threads.\n",
      " 69  TIM3e+I+G4    2030.606     38  4137.211     4140.530     4321.030\n",
      "WARNING: Number of threads seems too high for short alignments. Use -T AUTO to determine best number of threads.\n",
      " 70  TIM3e+R2      2031.377     38  4138.754     4142.073     4322.573\n",
      "WARNING: Number of threads seems too high for short alignments. Use -T AUTO to determine best number of threads.\n",
      " 79  TIM2+F        2021.280     39  4120.559     4124.057     4309.215\n",
      "WARNING: Number of threads seems too high for short alignments. Use -T AUTO to determine best number of threads.\n",
      " 80  TIM2+F+I      2017.995     40  4115.990     4119.671     4309.484\n",
      "WARNING: Number of threads seems too high for short alignments. Use -T AUTO to determine best number of threads.\n",
      " 81  TIM2+F+G4     2018.045     40  4116.090     4119.771     4309.583\n",
      "WARNING: Number of threads seems too high for short alignments. Use -T AUTO to determine best number of threads.\n",
      " 82  TIM2+F+I+G4   2018.027     41  4118.054     4121.924     4316.385\n",
      "WARNING: Number of threads seems too high for short alignments. Use -T AUTO to determine best number of threads.\n",
      " 83  TIM2+F+R2     2019.030     41  4120.060     4123.930     4318.391\n",
      "WARNING: Number of threads seems too high for short alignments. Use -T AUTO to determine best number of threads.\n",
      " 92  TIM2e         2033.539     36  4139.078     4142.055     4313.222\n",
      "WARNING: Number of threads seems too high for short alignments. Use -T AUTO to determine best number of threads.\n",
      " 93  TIM2e+I       2030.506     37  4135.012     4138.157     4313.993\n",
      "WARNING: Number of threads seems too high for short alignments. Use -T AUTO to determine best number of threads.\n",
      " 94  TIM2e+G4      2030.488     37  4134.976     4138.121     4313.957\n",
      "WARNING: Number of threads seems too high for short alignments. Use -T AUTO to determine best number of threads.\n",
      " 95  TIM2e+I+G4    2030.489     38  4136.977     4140.296     4320.796\n",
      "WARNING: Number of threads seems too high for short alignments. Use -T AUTO to determine best number of threads.\n",
      " 96  TIM2e+R2      2031.380     38  4138.760     4142.079     4322.578\n",
      "WARNING: Number of threads seems too high for short alignments. Use -T AUTO to determine best number of threads.\n",
      "105  TIM+F         2022.839     39  4123.677     4127.175     4312.333\n",
      "WARNING: Number of threads seems too high for short alignments. Use -T AUTO to determine best number of threads.\n",
      "106  TIM+F+I       2019.303     40  4118.606     4122.287     4312.099\n",
      "WARNING: Number of threads seems too high for short alignments. Use -T AUTO to determine best number of threads.\n",
      "107  TIM+F+G4      2019.393     40  4118.785     4122.467     4312.279\n",
      "WARNING: Number of threads seems too high for short alignments. Use -T AUTO to determine best number of threads.\n",
      "108  TIM+F+I+G4    2019.379     41  4120.758     4124.628     4319.089\n",
      "WARNING: Number of threads seems too high for short alignments. Use -T AUTO to determine best number of threads.\n",
      "109  TIM+F+R2      2020.505     41  4123.011     4126.880     4321.341\n",
      "WARNING: Number of threads seems too high for short alignments. Use -T AUTO to determine best number of threads.\n",
      "118  TIMe          2037.118     36  4146.236     4149.213     4320.380\n",
      "WARNING: Number of threads seems too high for short alignments. Use -T AUTO to determine best number of threads.\n",
      "119  TIMe+I        2034.044     37  4142.089     4145.234     4321.070\n",
      "WARNING: Number of threads seems too high for short alignments. Use -T AUTO to determine best number of threads.\n",
      "120  TIMe+G4       2034.073     37  4142.146     4145.292     4321.128\n",
      "WARNING: Number of threads seems too high for short alignments. Use -T AUTO to determine best number of threads.\n",
      "121  TIMe+I+G4     2034.064     38  4144.129     4147.448     4327.948\n",
      "WARNING: Number of threads seems too high for short alignments. Use -T AUTO to determine best number of threads.\n",
      "122  TIMe+R2       2034.957     38  4145.915     4149.234     4329.733\n",
      "WARNING: Number of threads seems too high for short alignments. Use -T AUTO to determine best number of threads.\n",
      "131  TPM3u+F       2020.612     38  4117.225     4120.544     4301.044\n",
      "WARNING: Number of threads seems too high for short alignments. Use -T AUTO to determine best number of threads.\n",
      "132  TPM3u+F+I     2017.164     39  4112.329     4115.826     4300.985\n",
      "WARNING: Number of threads seems too high for short alignments. Use -T AUTO to determine best number of threads.\n",
      "133  TPM3u+F+G4    2017.291     39  4112.582     4116.080     4301.238\n",
      "WARNING: Number of threads seems too high for short alignments. Use -T AUTO to determine best number of threads.\n",
      "134  TPM3u+F+I+G4  2017.267     40  4114.535     4118.216     4308.028\n",
      "WARNING: Number of threads seems too high for short alignments. Use -T AUTO to determine best number of threads.\n",
      "135  TPM3u+F+R2    2018.337     40  4116.675     4120.356     4310.168\n",
      "WARNING: Number of threads seems too high for short alignments. Use -T AUTO to determine best number of threads.\n",
      "144  TPM3+F        2020.610     38  4117.221     4120.540     4301.039\n",
      "WARNING: Number of threads seems too high for short alignments. Use -T AUTO to determine best number of threads.\n",
      "145  TPM3+F+I      2017.164     39  4112.327     4115.825     4300.983\n",
      "WARNING: Number of threads seems too high for short alignments. Use -T AUTO to determine best number of threads.\n",
      "146  TPM3+F+G4     2017.291     39  4112.582     4116.080     4301.238\n",
      "WARNING: Number of threads seems too high for short alignments. Use -T AUTO to determine best number of threads.\n",
      "147  TPM3+F+I+G4   2017.267     40  4114.534     4118.216     4308.028\n",
      "WARNING: Number of threads seems too high for short alignments. Use -T AUTO to determine best number of threads.\n",
      "148  TPM3+F+R2     2018.337     40  4116.675     4120.356     4310.168\n",
      "WARNING: Number of threads seems too high for short alignments. Use -T AUTO to determine best number of threads.\n",
      "157  TPM2u+F       2022.568     38  4121.135     4124.455     4304.954\n",
      "WARNING: Number of threads seems too high for short alignments. Use -T AUTO to determine best number of threads.\n",
      "158  TPM2u+F+I     2019.032     39  4116.063     4119.561     4304.719\n",
      "WARNING: Number of threads seems too high for short alignments. Use -T AUTO to determine best number of threads.\n",
      "159  TPM2u+F+G4    2019.120     39  4116.240     4119.738     4304.896\n",
      "WARNING: Number of threads seems too high for short alignments. Use -T AUTO to determine best number of threads.\n",
      "160  TPM2u+F+I+G4  2019.108     40  4118.216     4121.897     4311.710\n",
      "WARNING: Number of threads seems too high for short alignments. Use -T AUTO to determine best number of threads.\n",
      "161  TPM2u+F+R2    2020.248     40  4120.497     4124.178     4313.990\n",
      "WARNING: Number of threads seems too high for short alignments. Use -T AUTO to determine best number of threads.\n",
      "170  TPM2+F        2022.568     38  4121.135     4124.455     4304.954\n",
      "WARNING: Number of threads seems too high for short alignments. Use -T AUTO to determine best number of threads.\n",
      "171  TPM2+F+I      2019.032     39  4116.063     4119.561     4304.719\n",
      "WARNING: Number of threads seems too high for short alignments. Use -T AUTO to determine best number of threads.\n",
      "172  TPM2+F+G4     2019.120     39  4116.240     4119.738     4304.896\n",
      "WARNING: Number of threads seems too high for short alignments. Use -T AUTO to determine best number of threads.\n",
      "173  TPM2+F+I+G4   2019.108     40  4118.216     4121.897     4311.710\n",
      "WARNING: Number of threads seems too high for short alignments. Use -T AUTO to determine best number of threads.\n",
      "174  TPM2+F+R2     2020.248     40  4120.497     4124.178     4313.990\n",
      "WARNING: Number of threads seems too high for short alignments. Use -T AUTO to determine best number of threads.\n",
      "183  K3Pu+F        2024.126     38  4124.253     4127.572     4308.071\n",
      "WARNING: Number of threads seems too high for short alignments. Use -T AUTO to determine best number of threads.\n",
      "184  K3Pu+F+I      2020.423     39  4118.845     4122.343     4307.501\n",
      "WARNING: Number of threads seems too high for short alignments. Use -T AUTO to determine best number of threads.\n",
      "185  K3Pu+F+G4     2020.546     39  4119.092     4122.590     4307.748\n",
      "WARNING: Number of threads seems too high for short alignments. Use -T AUTO to determine best number of threads.\n",
      "186  K3Pu+F+I+G4   2020.533     40  4121.067     4124.748     4314.560\n",
      "WARNING: Number of threads seems too high for short alignments. Use -T AUTO to determine best number of threads.\n",
      "187  K3Pu+F+R2     2021.754     40  4123.509     4127.190     4317.002\n",
      "WARNING: Number of threads seems too high for short alignments. Use -T AUTO to determine best number of threads.\n",
      "196  K3P           2038.729     35  4147.458     4150.270     4316.764\n",
      "WARNING: Number of threads seems too high for short alignments. Use -T AUTO to determine best number of threads.\n",
      "197  K3P+I         2035.388     36  4142.777     4145.753     4316.921\n",
      "WARNING: Number of threads seems too high for short alignments. Use -T AUTO to determine best number of threads.\n",
      "198  K3P+G4        2035.471     36  4142.943     4145.919     4317.087\n",
      "WARNING: Number of threads seems too high for short alignments. Use -T AUTO to determine best number of threads.\n",
      "199  K3P+I+G4      2035.455     37  4144.909     4148.055     4323.891\n",
      "WARNING: Number of threads seems too high for short alignments. Use -T AUTO to determine best number of threads.\n",
      "200  K3P+R2        2036.489     37  4146.977     4150.123     4325.959\n",
      "WARNING: Number of threads seems too high for short alignments. Use -T AUTO to determine best number of threads.\n",
      "209  TN+F          2025.399     38  4126.798     4130.117     4310.617\n",
      "WARNING: Number of threads seems too high for short alignments. Use -T AUTO to determine best number of threads.\n",
      "210  TN+F+I        2021.738     39  4121.475     4124.973     4310.131\n",
      "WARNING: Number of threads seems too high for short alignments. Use -T AUTO to determine best number of threads.\n",
      "211  TN+F+G4       2021.805     39  4121.611     4125.109     4310.267\n",
      "WARNING: Number of threads seems too high for short alignments. Use -T AUTO to determine best number of threads.\n",
      "212  TN+F+I+G4     2021.800     40  4123.601     4127.282     4317.094\n",
      "WARNING: Number of threads seems too high for short alignments. Use -T AUTO to determine best number of threads.\n",
      "213  TN+F+R2       2023.013     40  4126.026     4129.708     4319.520\n",
      "WARNING: Number of threads seems too high for short alignments. Use -T AUTO to determine best number of threads.\n",
      "222  TNe           2039.701     35  4149.402     4152.214     4318.708\n",
      "WARNING: Number of threads seems too high for short alignments. Use -T AUTO to determine best number of threads.\n",
      "223  TNe+I         2036.503     36  4145.006     4147.983     4319.150\n",
      "WARNING: Number of threads seems too high for short alignments. Use -T AUTO to determine best number of threads.\n",
      "224  TNe+G4        2036.519     36  4145.038     4148.014     4319.181\n",
      "WARNING: Number of threads seems too high for short alignments. Use -T AUTO to determine best number of threads.\n",
      "225  TNe+I+G4      2036.515     37  4147.029     4150.174     4326.010\n",
      "WARNING: Number of threads seems too high for short alignments. Use -T AUTO to determine best number of threads.\n",
      "226  TNe+R2        2037.492     37  4148.983     4152.129     4327.965\n",
      "WARNING: Number of threads seems too high for short alignments. Use -T AUTO to determine best number of threads.\n",
      "235  HKY+F         2026.676     37  4127.352     4130.498     4306.334\n",
      "WARNING: Number of threads seems too high for short alignments. Use -T AUTO to determine best number of threads.\n",
      "236  HKY+F+I       2022.811     38  4121.622     4124.941     4305.441\n",
      "WARNING: Number of threads seems too high for short alignments. Use -T AUTO to determine best number of threads.\n",
      "237  HKY+F+G4      2022.914     38  4121.827     4125.146     4305.646\n",
      "WARNING: Number of threads seems too high for short alignments. Use -T AUTO to determine best number of threads.\n",
      "238  HKY+F+I+G4    2022.924     39  4123.848     4127.346     4312.504\n",
      "WARNING: Number of threads seems too high for short alignments. Use -T AUTO to determine best number of threads.\n",
      "239  HKY+F+R2      2024.241     39  4126.483     4129.980     4315.139\n",
      "WARNING: Number of threads seems too high for short alignments. Use -T AUTO to determine best number of threads.\n",
      "248  K2P           2041.307     34  4150.614     4153.268     4315.084\n",
      "WARNING: Number of threads seems too high for short alignments. Use -T AUTO to determine best number of threads.\n",
      "249  K2P+I         2037.818     35  4145.636     4148.449     4314.943\n",
      "WARNING: Number of threads seems too high for short alignments. Use -T AUTO to determine best number of threads.\n",
      "250  K2P+G4        2037.884     35  4145.769     4148.581     4315.075\n",
      "WARNING: Number of threads seems too high for short alignments. Use -T AUTO to determine best number of threads.\n",
      "251  K2P+I+G4      2037.881     36  4147.763     4150.739     4321.907\n",
      "WARNING: Number of threads seems too high for short alignments. Use -T AUTO to determine best number of threads.\n",
      "252  K2P+R2        2039.007     36  4150.013     4152.990     4324.157\n",
      "WARNING: Number of threads seems too high for short alignments. Use -T AUTO to determine best number of threads.\n",
      "261  F81+F         2063.985     36  4199.969     4202.946     4374.113\n",
      "WARNING: Number of threads seems too high for short alignments. Use -T AUTO to determine best number of threads.\n",
      "262  F81+F+I       2061.527     37  4197.053     4200.199     4376.035\n",
      "WARNING: Number of threads seems too high for short alignments. Use -T AUTO to determine best number of threads.\n",
      "263  F81+F+G4      2061.652     37  4197.304     4200.449     4376.285\n",
      "WARNING: Number of threads seems too high for short alignments. Use -T AUTO to determine best number of threads.\n",
      "264  F81+F+I+G4    2061.574     38  4199.149     4202.468     4382.967\n",
      "WARNING: Number of threads seems too high for short alignments. Use -T AUTO to determine best number of threads.\n",
      "265  F81+F+R2      2062.128     38  4200.255     4203.574     4384.074\n",
      "WARNING: Number of threads seems too high for short alignments. Use -T AUTO to determine best number of threads.\n",
      "274  JC            2077.652     33  4221.305     4223.803     4380.937\n",
      "WARNING: Number of threads seems too high for short alignments. Use -T AUTO to determine best number of threads.\n",
      "275  JC+I          2075.381     34  4218.762     4221.415     4383.231\n",
      "WARNING: Number of threads seems too high for short alignments. Use -T AUTO to determine best number of threads.\n",
      "276  JC+G4         2075.489     34  4218.978     4221.631     4383.447\n",
      "WARNING: Number of threads seems too high for short alignments. Use -T AUTO to determine best number of threads.\n",
      "277  JC+I+G4       2075.412     35  4220.824     4223.637     4390.131\n",
      "WARNING: Number of threads seems too high for short alignments. Use -T AUTO to determine best number of threads.\n",
      "278  JC+R2         2075.883     35  4221.767     4224.579     4391.074\n",
      "Akaike Information Criterion:           GTR+F+I\n",
      "Corrected Akaike Information Criterion: TVM+F+I\n",
      "Bayesian Information Criterion:         TPM3+F+I\n",
      "Best-fit model: TPM3+F+I chosen according to BIC\n",
      "\n",
      "All model information printed to output/FeLV-B.mafft.selectA1.aln.fasta.model.gz\n",
      "CPU time for ModelFinder: 59.401 seconds (0h:0m:59s)\n",
      "Wall-clock time for ModelFinder: 30.997 seconds (0h:0m:30s)\n",
      "Generating 1000 samples for ultrafast bootstrap (seed: 662094)...\n",
      "\n",
      "NOTE: 0 MB RAM (0 GB) is required!\n",
      "WARNING: Number of threads seems too high for short alignments. Use -T AUTO to determine best number of threads.\n",
      "Estimate model parameters (epsilon = 0.100)\n",
      "1. Initial log-likelihood: -2017.164\n",
      "Optimal log-likelihood: -2017.164\n",
      "Rate parameters:  A-C: 3.51598  A-G: 11.18505  A-T: 1.00000  C-G: 3.51598  C-T: 11.18505  G-T: 1.00000\n",
      "Base frequencies:  A: 0.314  C: 0.244  G: 0.206  T: 0.236\n",
      "Proportion of invariable sites: 0.549\n",
      "Parameters optimization took 1 rounds (0.001 sec)\n",
      "Computing ML distances based on estimated model parameters...\n",
      "Computing ML distances took 0.000843 sec (of wall-clock time) 0.001653 sec(of CPU time)\n",
      "Computing RapidNJ tree took 0.000125 sec (of wall-clock time) 0.000213 sec (of CPU time)\n",
      "Log-likelihood of RapidNJ tree: -2038.229\n",
      "--------------------------------------------------------------------\n",
      "|             INITIALIZING CANDIDATE TREE SET                      |\n",
      "--------------------------------------------------------------------\n",
      "Generating 98 parsimony trees... 0.034 second\n",
      "Computing log-likelihood of 98 initial trees ... 0.028 seconds\n",
      "Current best score: -2017.164\n",
      "\n",
      "Do NNI search on 20 best initial trees\n",
      "Estimate model parameters (epsilon = 0.100)\n",
      "BETTER TREE FOUND at iteration 1: -2017.163\n",
      "BETTER TREE FOUND at iteration 2: -2017.162\n",
      "BETTER TREE FOUND at iteration 3: -2017.162\n",
      "Iteration 10 / LogL: -2017.162 / Time: 0h:0m:0s\n",
      "Iteration 20 / LogL: -2017.163 / Time: 0h:0m:0s\n",
      "Finish initializing candidate tree set (17)\n",
      "Current best tree score: -2017.162 / CPU time: 0.155\n",
      "Number of iterations: 20\n",
      "--------------------------------------------------------------------\n",
      "|               OPTIMIZING CANDIDATE TREE SET                      |\n",
      "--------------------------------------------------------------------\n",
      "Iteration 30 / LogL: -2017.164 / Time: 0h:0m:0s (0h:0m:0s left)\n",
      "Iteration 40 / LogL: -2017.169 / Time: 0h:0m:0s (0h:0m:0s left)\n",
      "Iteration 50 / LogL: -2017.164 / Time: 0h:0m:0s (0h:0m:0s left)\n",
      "Log-likelihood cutoff on original alignment: -2029.514\n",
      "Iteration 60 / LogL: -2017.163 / Time: 0h:0m:0s (0h:0m:0s left)\n",
      "Iteration 70 / LogL: -2017.162 / Time: 0h:0m:0s (0h:0m:0s left)\n",
      "Iteration 80 / LogL: -2017.162 / Time: 0h:0m:0s (0h:0m:0s left)\n",
      "Iteration 90 / LogL: -2017.164 / Time: 0h:0m:0s (0h:0m:0s left)\n",
      "Iteration 100 / LogL: -2017.164 / Time: 0h:0m:0s (0h:0m:0s left)\n",
      "Log-likelihood cutoff on original alignment: -2033.936\n",
      "NOTE: Bootstrap correlation coefficient of split occurrence frequencies: 0.988\n",
      "NOTE: UFBoot does not converge, continue at least 100 more iterations\n",
      "Iteration 110 / LogL: -2017.166 / Time: 0h:0m:1s (0h:0m:0s left)\n",
      "UPDATE BEST LOG-LIKELIHOOD: -2017.162\n",
      "Iteration 120 / LogL: -2017.164 / Time: 0h:0m:1s (0h:0m:0s left)\n",
      "Iteration 130 / LogL: -2017.165 / Time: 0h:0m:1s (0h:0m:0s left)\n",
      "Iteration 140 / LogL: -2022.958 / Time: 0h:0m:1s (0h:0m:0s left)\n",
      "Iteration 150 / LogL: -2017.163 / Time: 0h:0m:1s (0h:0m:0s left)\n",
      "Log-likelihood cutoff on original alignment: -2035.387\n",
      "Iteration 160 / LogL: -2022.800 / Time: 0h:0m:1s (0h:0m:0s left)\n",
      "Iteration 170 / LogL: -2022.932 / Time: 0h:0m:1s (0h:0m:0s left)\n",
      "Iteration 180 / LogL: -2022.797 / Time: 0h:0m:1s (0h:0m:0s left)\n",
      "Iteration 190 / LogL: -2017.165 / Time: 0h:0m:1s (0h:0m:0s left)\n",
      "Iteration 200 / LogL: -2017.164 / Time: 0h:0m:1s (0h:0m:0s left)\n",
      "Log-likelihood cutoff on original alignment: -2036.640\n",
      "NOTE: Bootstrap correlation coefficient of split occurrence frequencies: 0.992\n",
      "TREE SEARCH COMPLETED AFTER 200 ITERATIONS / Time: 0h:0m:1s\n",
      "\n",
      "--------------------------------------------------------------------\n",
      "|                    FINALIZING TREE SEARCH                        |\n",
      "--------------------------------------------------------------------\n",
      "Performs final model parameters optimization\n",
      "Estimate model parameters (epsilon = 0.010)\n",
      "1. Initial log-likelihood: -2017.162\n",
      "Optimal log-likelihood: -2017.162\n",
      "Rate parameters:  A-C: 3.52228  A-G: 11.20639  A-T: 1.00000  C-G: 3.52228  C-T: 11.20639  G-T: 1.00000\n",
      "Base frequencies:  A: 0.314  C: 0.244  G: 0.206  T: 0.236\n",
      "Proportion of invariable sites: 0.549\n",
      "Parameters optimization took 1 rounds (0.001 sec)\n",
      "BEST SCORE FOUND : -2017.162\n",
      "Creating bootstrap support values...\n",
      "Split supports printed to NEXUS file output/FeLV-B.mafft.selectA1.aln.fasta.splits.nex\n",
      "Total tree length: 0.222\n",
      "\n",
      "Total number of iterations: 200\n",
      "CPU time used for tree search: 3.831 sec (0h:0m:3s)\n",
      "Wall-clock time used for tree search: 1.864 sec (0h:0m:1s)\n",
      "Total CPU time used: 3.949 sec (0h:0m:3s)\n",
      "Total wall-clock time used: 1.926 sec (0h:0m:1s)\n",
      "\n",
      "Computing bootstrap consensus tree...\n",
      "Reading input file output/FeLV-B.mafft.selectA1.aln.fasta.splits.nex...\n",
      "18 taxa and 216 splits.\n",
      "Consensus tree written to output/FeLV-B.mafft.selectA1.aln.fasta.contree\n",
      "Reading input trees file output/FeLV-B.mafft.selectA1.aln.fasta.contree\n",
      "Log-likelihood of consensus tree: -2017.176\n",
      "\n",
      "Analysis results written to: \n",
      "  IQ-TREE report:                output/FeLV-B.mafft.selectA1.aln.fasta.iqtree\n",
      "  Maximum-likelihood tree:       output/FeLV-B.mafft.selectA1.aln.fasta.treefile\n",
      "  Likelihood distances:          output/FeLV-B.mafft.selectA1.aln.fasta.mldist\n",
      "\n",
      "Ultrafast bootstrap approximation results written to:\n",
      "  Split support values:          output/FeLV-B.mafft.selectA1.aln.fasta.splits.nex\n",
      "  Consensus tree:                output/FeLV-B.mafft.selectA1.aln.fasta.contree\n",
      "  Screen log file:               output/FeLV-B.mafft.selectA1.aln.fasta.log\n",
      "\n",
      "Date and Time: Wed Dec 20 10:43:18 2023\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "[INFO]\u001b[0m read sequences ...\n",
      "[INFO]\u001b[0m 14 sequences loaded\n",
      "[INFO]\u001b[0m sorting ...\n",
      "[INFO]\u001b[0m output ...\n",
      "nadd =  10\n",
      "npair =  140\n",
      "nseq =  24\n",
      "nlen =  2280\n",
      "use multipair, weighti=2.7!\n",
      "nadd = 10\n",
      "generating a scoring matrix for nucleotide (dist=200) ... done\n",
      "   1/  14\r",
      "   2/  14\r",
      "   3/  14\r",
      "   4/  14\r",
      "   5/  14\r",
      "   6/  14\r",
      "   7/  14\r",
      "   8/  14\r",
      "   9/  14\r",
      "  10/  14\r",
      "  11/  14\r",
      "  12/  14\r",
      "  13/  14\r",
      "dndpre (nuc) Version 7.407\n",
      "alg=X, model=DNA200 (2), 2.00 (6.00), -0.10 (-0.30), noshift, amax=0.0\n",
      "0 thread(s)\n",
      "\n",
      "generating a scoring matrix for nucleotide (dist=200) ... done\n",
      "All-to-all alignment.\n",
      "    0 / 14\r",
      "    1 / 14\r",
      "    2 / 14\r",
      "    3 / 14\r",
      "    4 / 14\r",
      "    5 / 14\r",
      "    6 / 14\r",
      "    7 / 14\r",
      "    8 / 14\r",
      "    9 / 14\r",
      "   10 / 14\r",
      "   11 / 14\r",
      "   12 / 14\r",
      "   13 / 14\r\n",
      "\n",
      "##### writing hat3\n",
      "pairlocalalign (nuc) Version 7.407\n",
      "alg=Y, model=DNA200 (2), 2.00 (6.00), -0.10 (-0.30), noshift, amax=0.0\n",
      "0 thread(s)\n",
      "\n",
      "nadd = 10\n",
      "ppenalty_ex = -10\n",
      "nthread = 0\n",
      "blosum 62 / kimura 200\n",
      "sueff_global = 0.100000\n",
      "norg = 14\n",
      "njobc = 15\n",
      "Loading 'hat3' ... \n",
      "done.\n",
      "generating a scoring matrix for nucleotide (dist=200) ... done\n",
      "Loading 'hat2n' (aligned sequences - new sequences) ... done.\n",
      "Loading 'hat2i' (aligned sequences) ... done.\n",
      "\r",
      "0 / 10                    \r",
      "WARNING: negative branch length -0.086410, step 13-0\n",
      "c\r",
      "1 / 10                    \r",
      "WARNING: negative branch length -0.086410, step 13-0\n",
      "c\r",
      "2 / 10                    \r",
      "WARNING: negative branch length -0.108607, step 13-0\n",
      "c\r",
      "3 / 10                    \r",
      "WARNING: negative branch length -0.097361, step 13-0\n",
      "c\r",
      "4 / 10                    \r",
      "WARNING: negative branch length -0.104571, step 13-0\n",
      "c\r",
      "5 / 10                    \r",
      "WARNING: negative branch length -0.097361, step 13-0\n",
      "c\r",
      "6 / 10                    \r",
      "WARNING: negative branch length -0.086410, step 13-0\n",
      "c\r",
      "7 / 10                    \r",
      "WARNING: negative branch length -0.104792, step 13-0\n",
      "c\r",
      "8 / 10                    \r",
      "WARNING: negative branch length -0.104571, step 13-0\n",
      "c\r",
      "9 / 10                    \r",
      "WARNING: negative branch length -0.086410, step 13-0\n",
      "c\n",
      "\n",
      "Combining ..\n",
      "0 / 10\r",
      " 0 / 10\r",
      "\r",
      "   done.                      \n",
      "\n",
      "\r",
      "   done.                      \n",
      "\n",
      "addsingle (nuc) Version 7.407\n",
      "alg=A, model=DNA200 (2), 1.53 (4.59), -0.00 (-0.00), noshift, amax=0.0\n",
      "0 thread(s)\n",
      "\n",
      "\n",
      "To keep the alignment length, 13042 letters were DELETED.\n",
      "To know the positions of deleted letters, rerun the same command with the --mapout option.\n",
      "\n",
      "Strategy:\n",
      " Multi-INS-fragment (Not tested.)\n",
      " ?\n",
      "\n",
      "If unsure which option to use, try 'mafft --auto input > output'.\n",
      "For more information, see 'mafft --help', 'mafft --man' and the mafft page.\n",
      "\n",
      "The default gap scoring scheme has been changed in version 7.110 (2013 Oct).\n",
      "It tends to insert more gaps into gap-rich regions than previous versions.\n",
      "To disable this change, add the --leavegappyregion option.\n",
      "\n",
      "OMP: Info #273: omp_set_nested routine deprecated, please use omp_set_max_active_levels instead.\n",
      "OMP: Info #273: omp_set_nested routine deprecated, please use omp_set_max_active_levels instead.\n",
      "OMP: Info #273: omp_set_nested routine deprecated, please use omp_set_max_active_levels instead.\n"
     ]
    }
   ],
   "source": [
    "%%bash\n",
    "\n",
    "cat output/FeLV-B.mafft.smplot.fasta | \\\n",
    "    seqkit subseq -r 1619:2550 | \\\n",
    "    seqkit grep -n -r -p \"AB674444\" -v | \\\n",
    "    seqkit grep -n -r -p \"AY364318.1\" -v | \\\n",
    "    seqkit sort --by-name --reverse  \\\n",
    "    > output/FeLV-B.mafft.selectA1.fasta\n",
    "\n",
    "mafft --auto --keeplength \\\n",
    "    --addfragments data/FeLV-A.fasta \\\n",
    "    output/FeLV-B.mafft.selectA1.fasta \\\n",
    "    > output/FeLV-B.mafft.selectA1.aln.fasta\n",
    "    \n",
    "iqtree \\\n",
    "    -s output/FeLV-B.mafft.selectA1.aln.fasta \\\n",
    "    -m TESTNEW \\\n",
    "    -redo \\\n",
    "    -bb 1000 \\\n",
    "    -nt AUTO"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "env_210819",
   "language": "python",
   "name": "env_210819"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.9.13"
  },
  "toc": {
   "base_numbering": 1,
   "nav_menu": {},
   "number_sections": true,
   "sideBar": true,
   "skip_h1_title": false,
   "title_cell": "Table of Contents",
   "title_sidebar": "Contents",
   "toc_cell": true,
   "toc_position": {},
   "toc_section_display": true,
   "toc_window_display": true
  }
 },
 "nbformat": 4,
 "nbformat_minor": 4
}
